### PR TITLE
Update config for RPC communication - Closes #6613

### LINKED ISSUE #6613: Update config for RPC communication
### Description

Update the Configuration to improve and handle multiple mode of RPC communication.

```ts
// Current Config
rpc: {
	mode: ipc | ws,
 	enable: true,
	port: 8080,
}

// In Bus constructor
// Problem: It starts IPC server anyways even if ipc mode is not chosen
if (this.config.rpc.enable) {
            this._ipcServer = new IPCServer({
                socketsDir: this.config.socketsPath.root,
                name: 'bus',
            });
  }

if (this.config.rpc.enable && this.config.rpc.mode === 'ws') {
			this._wsServer = new WSServer({
				path: '/ws',
				port: config.rpc.port,
				host: config.rpc.host,
				logger: this.logger,
			});
		}
```

```ts
// New Config
rpc: {
	ipc: { path: ‘tmp/external/’},
 	ws: { port: 8080 },
        modes: [‘ipc’, ‘ws’]
}

// In Bus constructor
if (this.config.rpc.ipc) {
            this._ipcServer = new IPCServer({
                socketsDir: this.config.socketsPath.root,
                name: 'bus',
            });
  }

if (this.config.rpc.ws) {
            this._wsServer = new WSServer({
                path: '/ws',
                port: config.rpc.port,
                host: config.rpc.host,
                logger: this.logger,
            });
  }
```

### Acceptance Criteria

- Config related tests should pass
- Should be able to run current `ws`, `ipc` modes concurrently

### Additional Information

Related #6612

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #6613

### How was it solved?

- Updated the configuration schema 
- Updated the usage of new configuration  

### How was it tested?

- Run unit tests

### CODE REVIEW

File: framework/src/controller/bus.ts [hunk 1]
```diff
@@ -313,7 +308,7 @@ export class Bus {
 		this._emitter.emit(eventName, notification);
 
 		// Communicate through unix socket
-		if (this.config.rpc.enable) {
+		if (this.config.rpc.modes.includes(RPC_MODES.IPC)) {
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply framework/src/controller/bus.ts:311]: I believe that can be address when we introduce that public/private ipc servers. Plugins will always communicate through private ipc server.

File: framework/src/schema/application_config_schema.ts [hunk 2]
```diff
@@ -370,10 +379,12 @@ export const applicationConfigSchema = {
 			logFileName: 'lisk.log',
 		},
 		rpc: {
-			enable: false,
-			mode: 'ipc',
-			port: 8080,
-			host: '127.0.0.1',
+			modes: ['ipc'],
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply framework/src/schema/application_config_schema.ts:382]: For now socket directory is created based on data path which is computed from the rootPath and label. Once we introduce the internal/external ipc server then this config will point to external ipc server and we can decide for having a default path for it other than data path. But even then I would suggest to keep the default socket directory for for external ipc server within the data path directory.

File: framework/src/controller/bus.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/controller/bus.ts:311]: yeah we can also do it later after controller issue. But internal IPC server might not be there if there is no plugin loaded in child process.

**Review by ishantiw — APPROVED** [review]

File: framework/src/schema/application_config_schema.ts [hunk 3]
```diff
@@ -338,22 +338,31 @@ export const applicationConfigSchema = {
...
+					properties: {
+						path: { type: 'string' },
+					},
 				},
-				host: {
-					type: 'string',
-					format: 'ip',
+				ws: {
+					type: 'object',
+					required: ['host', 'port', 'path'],
```
Comment (shuse2) [inline framework/src/schema/application_config_schema.ts:357]: if we have `default` values, should it be `required`?

Comment (nazarhussain) [reply framework/src/schema/application_config_schema.ts:357]: In terms of `schema` and validation I think it must be required. Our validator does not support `default` keyword, we are doing `mergeDeep` on application level, so having restrict schema definition is better.

**Review by shuse2 — APPROVED** [review]

**Pull request merged**